Paintable Coated Variant Of The Exhaust Pipe.

DangerousMilk opened the pull request:
**Paintable full-block version of the exhaust pipe.**

Tried my best to make a texture, please let me know if I need to change it.
Added US, UA, RU, DE and CH localization.

Also please tell me if I need to change anything because it's my first time working on minecraft mods and commiting to an open source project in general.

I hope the coding is not too bad on this.

![2025-06-15_20 09 02](https://github.com/user-attachments/assets/27c8309e-6142-48ad-b088-5ca33b452e01)
![2025-06-15_21 44 29](https://github.com/user-attachments/assets/701ba457-5ad0-4908-a6d5-643faad9d52b)

DangerousMilk committed:
Added paintable coated variant of the exhaust pipe.

DangerousMilk commented:
Wait I actually have an idea as to how to make it a bit cleaner. Will commit tomorrow.

wiesenmann commented:
a recipe would be neat aswell

DangerousMilk commented:
Oh, right, I forgot. Will add it in the next commit aswell when I get home

DangerousMilk committed:
Added recipe and changed function order to make more sense.

DangerousMilk commented:
Also do I need to make any changes to the code? Since I copied most of the functions from the paintable fluid duct because I couldn't figure out how to make the enclosed exhaust pipe inherit from it without overwriting all of the functions anyway. Or is this fine?

HbmMods commented:
Looks good, although you'll probably have to register the tile entity in `TileMappings`, the duct's tile won't save to disk otherwise.

DangerousMilk committed:
Added the TileEntity to the TileMappings. Removed unnecessary canConnectTo function and TileEntityPipePaintable import

DangerousMilk commented:
Okay so I added it to the TileMappings put it still won't save for some reason. I think it may be because the paintable fluid duct tile entity inherits from TileEntityPipeBaseNT and the paintable exhaust pipe inherits from ExhaustPipe which doesn't have some needed save functionality maybe?

HbmMods commented:
You should check the log file, it usually prints an error when there's an exception thrown or a missing mapping that would prevent the tile from saving.

DangerousMilk commented:
So there are no errors but I think the nbt functions are just not being called for some reason. I tested the nbt functions with print statements and they didn't print :(

<details>
  <summary>Logs</summary>
  
  ### Log
  OpenJDK 64-Bit Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release
  [17:16:05] [main/INFO] [GradleStart]: Extra: []
  [17:16:05] [main/INFO] [GradleStart]: Found and added coremod: codechicken.core.launch.CodeChickenCorePlugin
  [17:16:05] [main/INFO] [GradleStart]: Found and added coremod: codechicken.nei.asm.NEICorePlugin
  [17:16:05] [main/INFO] [GradleStart]: Found and added coremod: invtweaks.forge.asm.FMLPlugin
  [17:16:05] [main/INFO] [GradleStart]: Running with arguments: [--userProperties, {}, --assetsDir, /home/realfancy/.gradle/caches/minecraft/assets, --assetIndex, 1.7.10, --accessToken, {REDACTED}, --version, 1.7.10, --tweakClass, cpw.mods.fml.common.launcher.FMLTweaker, --tweakClass, net.minecraftforge.gradle.tweakers.CoremodTweaker]
  [17:16:05] [main/INFO] [LaunchWrapper]: Loading tweak class name cpw.mods.fml.common.launcher.FMLTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Using primary tweak class name cpw.mods.fml.common.launcher.FMLTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.gradle.tweakers.CoremodTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.common.launcher.FMLTweaker
  [17:16:05] [main/INFO] [FML]: Forge Mod Loader version 7.99.40.1614 for Minecraft 1.7.10 loading
  [17:16:05] [main/INFO] [FML]: Java is OpenJDK 64-Bit Server VM, version 1.8.0_452, running on Linux:amd64:6.8.0-60-generic, installed at /usr/lib/jvm/java-8-openjdk-amd64/jre
  [17:16:05] [main/INFO] [FML]: Managed to load a deobfuscated Minecraft name- we are in a deobfuscated environment. Skipping runtime deobfuscation
  [17:16:05] [main/INFO] [FML]: Found a command line coremod : codechicken.nei.asm.NEICorePlugin
  [17:16:05] [main/WARN] [FML]: The coremod codechicken.nei.asm.NEICorePlugin does not have a MCVersion annotation, it may cause issues with this version of Minecraft
  [17:16:05] [main/INFO] [FML]: Found a command line coremod : invtweaks.forge.asm.FMLPlugin
  [17:16:05] [main/WARN] [FML]: The coremod invtweaks.forge.asm.FMLPlugin does not have a MCVersion annotation, it may cause issues with this version of Minecraft
  [17:16:05] [main/INFO] [FML]: Found a command line coremod : codechicken.core.launch.CodeChickenCorePlugin
  [17:16:05] [main/WARN] [FML]: The coremod codechicken.core.launch.CodeChickenCorePlugin does not have a MCVersion annotation, it may cause issues with this version of Minecraft
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.gradle.tweakers.CoremodTweaker
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod cpw.mods.fml.relauncher.FMLCorePlugin
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod net.minecraftforge.classloading.FMLForgePlugin
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod codechicken.core.asm.MCPDeobfuscationTransformer.LoadPlugin
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod codechicken.nei.asm.NEICorePlugin
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod invtweaks.forge.asm.FMLPlugin
  [17:16:05] [main/INFO] [GradleStart]: Injecting location in coremod codechicken.core.launch.CodeChickenCorePlugin
  [17:16:05] [main/INFO] [LaunchWrapper]: Loading tweak class name cpw.mods.fml.common.launcher.FMLInjectionAndSortingTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Loading tweak class name cpw.mods.fml.common.launcher.FMLDeobfTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.gradle.tweakers.AccessTransformerTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.common.launcher.FMLInjectionAndSortingTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.common.launcher.FMLInjectionAndSortingTweaker
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/ERROR] [FML]: The binary patch set is missing. Either you are in a development environment, or things are not going to work!
  [17:16:05] [main/ERROR] [FML]: FML appears to be missing any signature data. This is not a good thing
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.relauncher.CoreModManager$FMLPluginWrapper
  [17:16:05] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.common.launcher.FMLDeobfTweaker
  [17:16:06] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.gradle.tweakers.AccessTransformerTweaker
  [17:16:06] [main/INFO] [GradleStart]: Remapping AccessTransformer rules...
  [17:16:06] [main/INFO] [LaunchWrapper]: Loading tweak class name cpw.mods.fml.common.launcher.TerminalTweaker
  [17:16:06] [main/INFO] [LaunchWrapper]: Calling tweak class cpw.mods.fml.common.launcher.TerminalTweaker
  [17:16:06] [main/INFO] [LaunchWrapper]: Launching wrapped minecraft {net.minecraft.client.main.Main}
  [17:16:07] [main/INFO]: Setting user: Player7
  [17:16:08] [Client thread/INFO]: LWJGL Version: 2.9.1
  [17:16:08] [Client thread/INFO] [STDOUT]: [tv.twitch.StandardCoreAPI:<init>:16]: If on Windows, make sure to provide all of the necessary dll's as specified in the twitchsdk README. Also, make sure to set the PATH environment variable to point to the directory containing the dll's.
  [17:16:08] [Client thread/ERROR]: Couldn't initialize twitch stream
  [17:16:08] [Client thread/INFO] [STDOUT]: [cpw.mods.fml.client.SplashProgress:start:188]: ---- Minecraft Crash Report ----
  // I bet Cylons wouldn't have this problem.
  
  Time: 6/17/25 5:16 PM
  Description: Loading screen debug info
  
  This is just a prompt for computer specs to be printed. THIS IS NOT A ERROR
  
  
  A detailed walkthrough of the error, its code path and all known details is as follows:
  ---------------------------------------------------------------------------------------
  
  -- System Details --
  Details:
  	Minecraft Version: 1.7.10
  	Operating System: Linux (amd64) version 6.8.0-60-generic
  	Java Version: 1.8.0_452, Private Build
  	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Private Build
  	Memory: 832021744 bytes (793 MB) / 1037959168 bytes (989 MB) up to 1037959168 bytes (989 MB)
  	JVM Flags: 3 total; -Xincgc -Xmx1024M -Xms1024M
  	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
  	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
  	FML: 
  	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 550.144.03' Renderer: 'NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2'
  [17:16:08] [Client thread/INFO] [MinecraftForge]: Attempting early MinecraftForge initialization
  [17:16:08] [Client thread/INFO] [FML]: MinecraftForge v10.13.4.1614 Initialized
  [17:16:09] [Client thread/INFO] [FML]: Replaced 183 ore recipies
  [17:16:09] [Client thread/INFO] [MinecraftForge]: Completed early MinecraftForge initialization
  [17:16:09] [Client thread/INFO] [FML]: Found 0 mods from the command line. Injecting into mod discoverer
  [17:16:09] [Client thread/INFO] [FML]: Searching /home/realfancy/Dev/Minecraft/HBM-fork/eclipse/mods for mods
  [17:16:09] [Client thread/INFO] [FML]: Also searching /home/realfancy/Dev/Minecraft/HBM-fork/eclipse/mods/1.7.10 for mods
  [17:16:13] [Client thread/INFO] [FML]: Forge Mod Loader has identified 7 mods to load
  [17:16:13] [Client thread/INFO] [FML]: Attempting connection with missing mods [mcp, FML, Forge, NotEnoughItems, CodeChickenCore, hbm, inventorytweaks] at CLIENT
  [17:16:13] [Client thread/INFO] [FML]: Attempting connection with missing mods [mcp, FML, Forge, NotEnoughItems, CodeChickenCore, hbm, inventorytweaks] at SERVER
  [17:16:14] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Not Enough Items, FMLFileResourcePack:Hbm's Nuclear Tech, FMLFileResourcePack:Inventory Tweaks
  [17:16:14] [Client thread/INFO] [FML]: Processing ObjectHolder annotations
  [17:16:14] [Client thread/INFO] [FML]: Found 341 ObjectHolder annotations
  [17:16:14] [Client thread/INFO] [FML]: Identifying ItemStackHolder annotations
  [17:16:14] [Client thread/INFO] [FML]: Found 0 ItemStackHolder annotations
  [17:16:14] [Client thread/INFO] [FML]: Configured a dormant chunk cache size of 0
  [17:16:14] [Client thread/INFO] [HBM]: Let us celebrate the fact that the logger finally works again!
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.Gson.fromJson(Gson.java:815)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.Gson.fromJson(Gson.java:768)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.Gson.fromJson(Gson.java:717)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.Gson.fromJson(Gson.java:689)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at net.minecraftforge.common.ForgeVersion$1.run(ForgeVersion.java:94)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:374)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:183)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	at com.google.gson.Gson.fromJson(Gson.java:803)
  [17:16:14] [Forge Version Check/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:749]: 	... 4 more
  [17:16:14] [Client thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.Container
  [17:16:20] [Client thread/INFO] [HBM]: [QMAW] Reloading manual...
  [17:16:20] [Client thread/INFO] [HBM]: [QMAW] Error dissecting zip QMAWLoader.class: java.util.zip.ZipException: error in opening zip file
  [17:16:20] [Client thread/INFO] [HBM]: [QMAW] Exploring /home/realfancy/Dev/Minecraft/HBM-fork/src/main/resources/assets/hbm/manual
  [17:16:20] [Client thread/INFO] [HBM]: [QMAW] Found manual demo.json
  [17:16:20] [Client thread/INFO] [HBM]: [QMAW] Loaded 0 manual entries! (1ms)
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotTcAlloy to ingotAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for dustTcAlloy to dustAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for blockTcAlloy to blockAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotCdAlloy to ingotAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for blockCdAlloy to blockAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotBismuthBronze to ingotAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotArsenicBronze to ingotAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotPolymer to ingotAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for dustPolymer to dustAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for blockPolymer to blockAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotBakelite to ingotAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for dustBakelite to dustAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for blockBakelite to blockAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotLatex to ingotAnyRubber
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotRubber to ingotAnyRubber
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotPolycarbonate to ingotAnyHardPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for ingotPVC to ingotAnyHardPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for oiltar to anyTar
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for cracktar to anyTar
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for coaltar to anyTar
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for woodtar to anyTar
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateTripleTcAlloy to plateTripleAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateSextupleTcAlloy to plateSextupleAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for componentHeavyTcAlloy to componentHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelLightTcAlloy to barrelLightAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelHeavyTcAlloy to barrelHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverLightTcAlloy to receiverLightAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverHeavyTcAlloy to receiverHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateTripleCdAlloy to plateTripleAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateSextupleCdAlloy to plateSextupleAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for componentHeavyCdAlloy to componentHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelLightCdAlloy to barrelLightAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelHeavyCdAlloy to barrelHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverLightCdAlloy to receiverLightAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverHeavyCdAlloy to receiverHeavyAnyResistantAlloy
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateTripleBismuthBronze to plateTripleAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelLightBismuthBronze to barrelLightAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverLightBismuthBronze to receiverLightAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverHeavyBismuthBronze to receiverHeavyAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for plateTripleArsenicBronze to plateTripleAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for barrelLightArsenicBronze to barrelLightAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverLightArsenicBronze to receiverLightAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for receiverHeavyArsenicBronze to receiverHeavyAnyBismoidBronze
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for stockPolymer to stockAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for gripPolymer to gripAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for stockBakelite to stockAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for gripBakelite to gripAnyPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for stockPolycarbonate to stockAnyHardPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for gripPolycarbonate to gripAnyHardPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for stockPVC to stockAnyHardPlastic
  [17:16:20] [Client thread/INFO] [HBM]: Re-registration for gripPVC to gripAnyHardPlastic
  [17:16:21] [Client thread/INFO] [FML]: InvTweaks: invtweaks.InvTweaksObfuscation
  [17:16:21] [Client thread/INFO] [FML]: Applying holder lookups
  [17:16:21] [Client thread/INFO] [FML]: Holder lookups applied
  [17:16:21] [Client thread/INFO] [FML]: Injecting itemstacks
  [17:16:21] [Client thread/INFO] [FML]: Itemstack injection complete
  [17:16:21] [NTM Version Checker/INFO] [HBM]: Searching for new versions...
  [17:16:21] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:21] [NTM Version Checker/INFO] [HBM]: Found version 1.0.27 BETA (5357)
  [17:16:21] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: Starting up SoundSystem...
  [17:16:21] [NTM Version Checker/INFO] [HBM]: Version checker ended.
  [17:16:21] [Thread-7/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: Initializing LWJGL OpenAL
  [17:16:21] [Thread-7/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]:     (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
  [17:16:21] [Thread-7/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: OpenAL initialized.
  [17:16:21] [Client thread/INFO]: Created: 16x16 textures/blocks-atlas
  [17:16:21] [Client thread/INFO]: Created: 16x16 textures/items-atlas
  [17:16:21] [Client thread/ERROR] [CodeChickenCore]: Unable to do mod description scrolling due to lack of stencil buffer
  [17:16:21] [Client thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerEnchantment
  [17:16:21] [Client thread/INFO] [inventorytweaks]: InvTweaks: Configuration loaded.
  [17:16:21] [Client thread/INFO] [inventorytweaks]: Mod initialized
  [17:16:21] [Client thread/INFO] [FML]: Injecting itemstacks
  [17:16:21] [Client thread/INFO] [FML]: Itemstack injection complete
  [17:16:21] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:21] [Sound Library Loader/INFO]: Sound engine started
  [17:16:21] [Client thread/INFO] [HBM]: Starting recipe init!
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmPress.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmPress.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmBlastFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmBlastFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmShredder.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmShredder.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmSoldering.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmSoldering.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmChemplant.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmChemplant.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCombination.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCombination.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCrucible.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCrucible.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCentrifuge.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCentrifuge.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCrystallizer.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCrystallizer.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmFractions.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmFractions.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCracking.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCracking.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmReforming.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmReforming.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmHydrotreating.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmHydrotreating.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmLiquefactor.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmLiquefactor.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmSolidifier.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmSolidifier.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCoker.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCoker.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmPyrolysis.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmPyrolysis.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmBreeder.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmBreeder.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCyclotron.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCyclotron.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmHadronCollider.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmHadronCollider.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmFuelpool.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmFuelpool.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmMixer.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmMixer.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmIrradiation.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmIrradiation.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCompressor.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCompressor.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmElectrolyzerFluid.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmElectrolyzerFluid.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmElectrolyzerMetal.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmElectrolyzerMetal.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmArcWelder.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmArcWelder.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmRotaryFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmRotaryFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmExposureChamber.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmExposureChamber.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmParticleAccelerator.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmParticleAccelerator.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmAmmoPress.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmAmmoPress.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmAssembler.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmAssembler.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmAnvil.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmAnvil.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmPedestal.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmPedestal.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmChemicalPlant.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmChemicalPlant.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCrucibleSmelting.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCrucibleSmelting.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmCustomMachines.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmCustomMachines.json
  [17:16:21] [Client thread/INFO] [HBM]: No recipe file found, registering defaults for hbmArcFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: Writing template file _hbmArcFurnace.json
  [17:16:21] [Client thread/INFO] [HBM]: Finished recipe init!
  [17:16:21] [Client thread/INFO] [HBM]: No armor file found, registering defaults for _hbmArmor.json
  [17:16:21] [Client thread/INFO] [FML]: Forge Mod Loader has successfully loaded 7 mods
  [17:16:21] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Not Enough Items, FMLFileResourcePack:Hbm's Nuclear Tech, FMLFileResourcePack:Inventory Tweaks
  [17:16:21] [Client thread/INFO] [HBM]: [QMAW] Reloading manual...
  [17:16:21] [Client thread/INFO] [HBM]: [QMAW] Error dissecting zip QMAWLoader.class: java.util.zip.ZipException: error in opening zip file
  [17:16:21] [Client thread/INFO] [HBM]: [QMAW] Exploring /home/realfancy/Dev/Minecraft/HBM-fork/src/main/resources/assets/hbm/manual
  [17:16:21] [Client thread/INFO] [HBM]: [QMAW] Found manual demo.json
  [17:16:21] [Client thread/INFO] [HBM]: [QMAW] Loaded 0 manual entries! (0ms)
  [17:16:22] [Client thread/INFO]: Created: 1024x1024 textures/blocks-atlas
  [17:16:23] [Client thread/INFO]: Created: 1024x1024 textures/items-atlas
  [17:16:23] [Client thread/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:23] [Client thread/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: SoundSystem shutting down...
  [17:16:23] [Client thread/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:importantMessage:90]:     Author: Paul Lamb, www.paulscode.com
  [17:16:23] [Client thread/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:23] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:23] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: Starting up SoundSystem...
  [17:16:24] [Thread-10/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: Initializing LWJGL OpenAL
  [17:16:24] [Thread-10/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]:     (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
  [17:16:24] [Thread-10/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: OpenAL initialized.
  [17:16:24] [Sound Library Loader/INFO] [STDOUT]: [paulscode.sound.SoundSystemLogger:message:69]: 
  [17:16:24] [Sound Library Loader/INFO]: Sound engine started
  [17:16:27] [Server thread/INFO]: Starting integrated minecraft server version 1.7.10
  [17:16:27] [Server thread/INFO]: Generating keypair
  [17:16:27] [Server thread/INFO] [FML]: Injecting existing block and item data into this server instance
  [17:16:27] [Server thread/INFO] [FML]: Applying holder lookups
  [17:16:27] [Server thread/INFO] [FML]: Holder lookups applied
  [17:16:27] [Server thread/INFO] [FML]: Loading dimension 0 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@45b56680)
  [17:16:27] [Server thread/INFO] [FML]: Loading dimension 1 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@45b56680)
  [17:16:27] [Server thread/INFO] [FML]: Loading dimension -1 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@45b56680)
  [17:16:27] [Server thread/INFO]: Preparing start region for level 0
  [17:16:27] [Server thread/INFO] [HBM]: Attempting to re-register item stats...
  [17:16:27] [Server thread/INFO] [HBM]: Item stats re-registered
  [17:16:28] [Server thread/INFO]: Changing view distance to 12, from 10
  [17:16:28] [Client thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerPlayer
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerChest
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerHopper
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerRepair
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerWorkbench
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerFurnace
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerDispenser
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerBrewingStand
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerBeacon
  [17:16:28] [Server thread/INFO] [FML]: InvTweaks: net.minecraft.inventory.ContainerMerchant
  [17:16:28] [Netty Client IO #0/INFO] [FML]: Server protocol version 2
  [17:16:28] [Netty IO #1/INFO] [FML]: Client protocol version 2
  [17:16:28] [Netty IO #1/INFO] [FML]: Client attempting to join with 7 mods : NotEnoughItems@1.0.3.74,CodeChickenCore@1.0.4.29,hbm@1.0.27 BETA (5357),FML@7.10.99.99,inventorytweaks@1.59-dev-af3bc68-dirty,Forge@10.13.4.1614,mcp@9.05
  [17:16:28] [Netty IO #1/INFO] [FML]: Attempting connection with missing mods [] at CLIENT
  [17:16:28] [Netty Client IO #0/INFO] [FML]: Attempting connection with missing mods [] at SERVER
  [17:16:28] [Server thread/INFO] [FML]: [Server thread] Server side modded connection established
  [17:16:28] [Server thread/INFO]: Player7[local:E:71a8bfa4] logged in with entity id 169 at (120.7676384380197, 82.0, 92.70521286839902)
  [17:16:28] [Server thread/INFO]: Player7 joined the game
  [17:16:28] [Client thread/INFO] [FML]: [Client thread] Client side modded connection established
  [17:16:28] [Client thread/INFO]: [CHAT] Loaded world with Hbm's Nuclear Tech Mod 1.0.27 BETA (5357) for Minecraft 1.7.10!
  [17:16:30] [Client thread/INFO]: [CHAT] Version 1.0.7.48 of CodeChickenCore is available
  [17:16:30] [Client thread/INFO]: [CHAT] Version 1.0.5.120 of NotEnoughItems is available
  [17:16:30] [Client thread/INFO]: [CHAT] InvTweaks: Configuration loaded.
  [17:16:32] [Server thread/INFO]: Saving and pausing game...
  [17:16:33] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Overworld
  [17:16:33] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Nether
  [17:16:33] [Server thread/INFO]: Saving chunks for level 'New Worlda'/The End
  [17:21:31] [Server thread/INFO]: Saving and pausing game...
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Overworld
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Nether
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/The End
  [17:21:31] [Server thread/INFO]: Stopping server
  [17:21:31] [Server thread/INFO]: Saving players
  [17:21:31] [Server thread/INFO]: Saving worlds
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Overworld
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Nether
  [17:21:31] [Server thread/INFO]: Saving chunks for level 'New Worlda'/The End
  [17:21:32] [Server thread/INFO] [FML]: Unloading dimension 0
  [17:21:32] [Server thread/INFO] [FML]: Unloading dimension -1
  [17:21:32] [Server thread/INFO] [FML]: Unloading dimension 1
  [17:21:32] [Server thread/INFO] [FML]: Applying holder lookups
  [17:21:32] [Server thread/INFO] [FML]: Holder lookups applied
  [17:21:35] [Server thread/INFO]: Starting integrated minecraft server version 1.7.10
  [17:21:35] [Server thread/INFO]: Generating keypair
  [17:21:35] [Server thread/INFO] [FML]: Injecting existing block and item data into this server instance
  [17:21:35] [Server thread/INFO] [FML]: Applying holder lookups
  [17:21:35] [Server thread/INFO] [FML]: Holder lookups applied
  [17:21:35] [Server thread/INFO] [FML]: Loading dimension 0 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@91f1d9f)
  [17:21:35] [Server thread/INFO] [FML]: Loading dimension 1 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@91f1d9f)
  [17:21:35] [Server thread/INFO] [FML]: Loading dimension -1 (New Worlda) (net.minecraft.server.integrated.IntegratedServer@91f1d9f)
  [17:21:35] [Server thread/INFO]: Preparing start region for level 0
  [17:21:35] [Server thread/INFO] [HBM]: Attempting to re-register item stats...
  [17:21:35] [Server thread/INFO] [HBM]: Item stats re-registered
  [17:21:35] [Server thread/INFO]: Changing view distance to 12, from 10
  [17:21:35] [Netty Client IO #1/INFO] [FML]: Server protocol version 2
  [17:21:35] [Netty IO #3/INFO] [FML]: Client protocol version 2
  [17:21:35] [Netty IO #3/INFO] [FML]: Client attempting to join with 7 mods : NotEnoughItems@1.0.3.74,CodeChickenCore@1.0.4.29,hbm@1.0.27 BETA (5357),FML@7.10.99.99,inventorytweaks@1.59-dev-af3bc68-dirty,Forge@10.13.4.1614,mcp@9.05
  [17:21:35] [Netty IO #3/INFO] [FML]: Attempting connection with missing mods [] at CLIENT
  [17:21:35] [Netty Client IO #1/INFO] [FML]: Attempting connection with missing mods [] at SERVER
  [17:21:35] [Client thread/INFO] [FML]: [Client thread] Client side modded connection established
  [17:21:35] [Server thread/INFO] [FML]: [Server thread] Server side modded connection established
  [17:21:35] [Server thread/INFO]: Player7[local:E:6519bfa6] logged in with entity id 640 at (121.84062712633028, 82.13955412766411, 92.83560159609064)
  [17:21:35] [Server thread/INFO]: Player7 joined the game
  [17:21:36] [Client thread/INFO]: [CHAT] Loaded world with Hbm's Nuclear Tech Mod 1.0.27 BETA (5357) for Minecraft 1.7.10!
  [17:21:38] [Server thread/INFO]: Saving and pausing game...
  [17:21:38] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Overworld
  [17:21:38] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Nether
  [17:21:38] [Server thread/INFO]: Saving chunks for level 'New Worlda'/The End
  [17:31:47] [Server thread/INFO]: Saving and pausing game...
  [17:31:47] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Overworld
  [17:31:47] [Server thread/INFO]: Saving chunks for level 'New Worlda'/Nether
  [17:31:47] [Server thread/INFO]: Saving chunks for level 'New Worlda'/The End

</details>

wiesenmann commented:
`TileEntityPipeBaseNT` which `TileEntityPipePaintable` inherits from, does call the `writeToNBT` method inside of the `getDescriptionPacket` method and the `readFromNBT` method inside of the `onDataPacket` method. Your `TileEntityPipeExhaustPaintable` and `TileEntityPipeExhaust` which yours inherits from however do not call the `readFromNBT` and `writeFromNBT` methods inside the `getDescriptionPacket` and `onDataPacket` methods.

DangerousMilk committed:
Fixed NBT data not saving.

DangerousMilk commented:
Thanks, that should have fixed it! :D

wiesenmann commented:
Seems to be working now.
